### PR TITLE
Initial support for clangd in lsp.

### PR DESCRIPTION
**Pull request by tee3**
This has been tested by evaluating the code added and enabling lsp on a C++ project.

### CODE REVIEW

**Comment by yyoncho** [comment]
Thanks for porting this. I will delay the merge since the story when there are 2 servers handling one mode but the clients want only one of them is not implemented.

**Comment by tee3** [comment]
👍

File: lsp-clients.el [hunk 1]
```diff
@@ -235,5 +235,32 @@ PARAMS progress report notification data."
...
+
+(defcustom lsp-clients-clangd-args '()
+  "Extra arguments for the clangd executable."
+  :group 'lsp-clangd
+  :risky t
+  :type '(repeat string))
+
+(defun lsp-clients--clangd-command ()
+  "Generate the language server startup command."
+  (concat lsp-clients-clangd-executable (concat lsp-clients-clangd-args)))
```
Comment (yyoncho) [inline lsp-clients.el]: concat works for strings, here you need `(,lsp-clients-langd-executable ,@lsp-clients-clangd-args)

**Comment by tee3** [comment]
I've made the requested changes, I hope it is what you are expecting.

**Comment by yyoncho** [comment]
Thanks, Looks good, can you resolve the conflicts so I can merge it?

**Comment by tee3** [comment]
Should I resolve using rebase or merge?  Do you have a preference?

**Comment by yyoncho** [comment]
It doesn't matter for me. I guess in most of the project rebase is preferred.

**Comment by tee3** [comment]
Rebased and pushed.

**Comment by yyoncho** [comment]
Thanks!

**Pull request merged**